### PR TITLE
Compass encrypt

### PR DESCRIPTION
**Pull request by gabrielleitezup**
# CharlesCD Compass breaking change

## What have changed?

Compass now encryptes fields containing configuration data on the database, with that you won't be able to use the metric providers already registered.

## I need to register my providers again?

No, we wrote a script so you can use the providers that already had been registered.

## How the scripts works?

It's sql file that you need to execuce on your compass database. When executed the script will ask for a private key to encrypt your data(We recommend to use password generator).
The script is located on compass resources folder.

## Private key, what is that?

Now we use a private key to encrypt your data, you will need to set that as environment variable. Remember choose a strong key that you will utilize for both cases(Script end Environment Variable)

### CODE REVIEW

File: compass/.env [hunk 1]
```diff
@@ -6,4 +6,5 @@ DB_SSL=disable
 DB_PORT=5432
 ENV=DEV
 PLUGINS_DIR=./dist
-TIMEOUT=10
\ No newline at end of file
+TIMEOUT=10
+ENCRYPTION_KEY=MAYCON
```
Comment (maycommit) [inline compass/.env:10]: :)

File: compass/internal/action/action.go [hunk 2]
```diff
@@ -177,3 +226,22 @@ func (main Main) DeleteAction(id string) error {
 	}
 	return nil
 }
+
+func actionInsert(id, nickname, actionType, description string, config []byte, workspaceId uuid.UUID) string {
```
Comment (arthurpuccizup) [inline compass/internal/action/action.go:230]: Levar essa função para dentro do arquivo queries, acho que fica mais coeso as responsabilidades

File: compass/internal/configuration/configuration.go [hunk 3]
```diff
@@ -43,6 +43,7 @@ var initialValues = map[string]string{
 	"MOOVE_USER":          "admin@admin",
 	"MOOVE_PATH":          "http://charlescd-moove:8080",
 	"MOOVE_AUTH":          "Bearer 123",
+	"ENCRYPTION_KEY":      "MAYCON",
```
Comment (maycommit) [inline compass/internal/configuration/configuration.go:46]: :)

File: compass/internal/tests/util.go [hunk 4]
```diff
@@ -75,6 +78,24 @@ func newBasicDatasource() datasource.DataSource {
 	}
 }
 
+func datasourceInsert(pluginSrc string) (string, datasource.Response) {
```
Comment (maycommit) [inline compass/internal/tests/util.go:81]: Porque essa função nao esta no datasource internal

File: compass/internal/tests/util.go [hunk 5]
```diff
@@ -100,6 +121,36 @@ func newBasicAction() action.Action {
 	}
 }
 
+func actionInsert(actionType string) (string, action.Response) {
```
Comment (maycommit) [inline compass/internal/tests/util.go:124]: RT

File: compass/plugins/datasource/prometheus/prometheus.go [hunk 6]
```diff
@@ -133,6 +133,9 @@ func GetMetrics(datasourceConfiguration []byte) (datasource.MetricList, error) {
 func Query(request datasource.QueryRequest) ([]datasource.Value, error) {
 	apiClient, err := getPrometheusApiClient(request.DatasourceConfiguration)
 	if err != nil {
+		var prometheusConfig map[string]string
```
Comment (maycommit) [inline compass/plugins/datasource/prometheus/prometheus.go:136]: Porque esse decode para um printf?

File: compass/internal/datasource/datasource.go [hunk 7]
```diff
@@ -189,18 +231,44 @@ func (main Main) VerifyHealthAtWorkspace(workspaceId string) (bool, error) {
...
+	}
 
-	db := main.db.Create(&dataSource)
-	if db.Error != nil {
-		logger.Error(util.DatasourceSaveError, "SaveAction", db.Error, dataSource)
-		return DataSource{}, db.Error
+	return entity.toResponse(), nil
+}
+
+func datasourceInsert(id, name, pluginSrc string, data []byte, health bool, workspaceId uuid.UUID) string {
```
Comment (arthurpuccizup) [inline compass/internal/datasource/datasource.go:258]: Mover essa função para dentro do arquivo de queries

File: compass/plugins/datasource/prometheus/prometheus.go [hunk 8]
```diff
@@ -146,6 +149,9 @@ func Query(request datasource.QueryRequest) ([]datasource.Value, error) {
 
 	result, _, err := v1Api.Query(context.Background(), buildedQuery, time.Now())
 	if err != nil {
+		var prometheusConfig map[string]string
```
Comment (maycommit) [inline compass/plugins/datasource/prometheus/prometheus.go:152]: RT

**Review by maycommit — CHANGES_REQUESTED** [review]

**Review by arthurpuccizup — CHANGES_REQUESTED** [review]

File: compass/internal/tests/util.go (hunk 4 above)
Comment (gabrielleitezup) [reply compass/internal/tests/util.go:81]: Temos a função Insert no datasource internal, mas pra atender o os requisitos dos testes precisado retornar a query e o datasource com os dados predefenidos. Essa função soh server para ajudar nos testes.

File: compass/internal/tests/util.go (hunk 5 above)
Comment (gabrielleitezup) [reply compass/internal/tests/util.go:124]: Temos a função Insert no datasource internal, mas pra atender o os requisitos dos testes precisado retornar a query e o datasource com os dados predefenidos. Essa função soh server para ajudar nos testes.

**Review by maycommit — DISMISSED** [review]

**Review by maycommit — APPROVED** [review]

**Review by arthurpuccizup — APPROVED** [review]

**Review by arthurpuccizup — APPROVED** [review]

**Review by maycommit — APPROVED** [review]

**Pull request merged**